### PR TITLE
Skip namespacestores with invalid configuration only

### PR DESCRIPTION
**Pull request by igorpick**
In order to avoid the removing existing mount points when namespacestore gets the rejected state,
ignore the state and try to configure the mounts.

### CODE REVIEW

**Review by jeniawhite — APPROVED** [review]

File: pkg/system/phase4_configuring.go [hunk 1]
```diff
@@ -455,6 +455,37 @@ func (r *Reconciler) setDesiredEndpointMounts(podSpec *corev1.PodSpec, container
 	return nil
 }
 
+// Duplicate code from validation.go namespacetore pkg.
+// Cannot import the namespacestore pkg, because the pkg imports the system pkg
+// TODO remove the code
+func (r *Reconciler) validateNsStoreNSFS(nsStore *nbv1.NamespaceStore) bool {
+	nsfs := nsStore.Spec.NSFS
+
+	if nsfs == nil {
```
Comment (jackyalbo) [inline pkg/system/phase4_configuring.go:464]: I think this check is repeating itself - but as we want to remove this code later - I think it's fine

File: pkg/system/phase4_configuring.go [hunk 2]
```diff
@@ -409,8 +407,10 @@ func (r *Reconciler) setDesiredEndpointMounts(podSpec *corev1.PodSpec, container
 	container.VolumeMounts = r.DefaultDeploymentEndpoint.Containers[0].VolumeMounts
 
 	for _, nsStore := range namespaceStoreList.Items {
-		if nsStore.Status.Phase == nbv1.NamespaceStorePhaseRejected {
-			log.Warnf("namespacestore %s is in rejected phase", nsStore.Name)
```
Comment (jackyalbo) [inline pkg/system/phase4_configuring.go:413]: maybe leave this warning next to the continue - just change to namespace store is not valid. WDYT?

Comment (igorpick) [reply pkg/system/phase4_configuring.go:413]: You told me that you don't want to see this warning each 3 sec, no?

File: pkg/system/phase4_configuring.go (hunk 1 above)
Comment (igorpick) [reply pkg/system/phase4_configuring.go:464]: Could you explain me pls, why do you think it's repeating?

Comment (jackyalbo) [reply pkg/system/phase4_configuring.go:464]: OK, We will have the status in the namespacestore describe. leave it

**Review by jackyalbo — APPROVED** [review]

File: pkg/system/phase4_configuring.go (hunk 2 above)
Comment (jackyalbo) [reply pkg/system/phase4_configuring.go:413]: After our small discussion - it's fine we will get the reject reason from the namespacestore describe

**Pull request merged**